awx/main/tasks.py

```diff
@@ -1246,6 +1247,10 @@ def event_handler(self, event_data):
                 event_data.pop('parent_uuid', None)
         if self.parent_workflow_job_id:
             event_data['workflow_job_id'] = self.parent_workflow_job_id
+        # Do we have to check if the field exists? if it doesn't
+        # how will be eventually store the event in the db?
+        if self.job_created:
```

Comment:
Yep, I'd say we don't want this if statement, because if this value isn't set, the event won't be saved (because it's what the partitioning is based on).

If you get to this part of the codebase and `self.job_created` isn't set, there's a bug (so let's expose it).

Reply:
makes sense - dropped comment / conditional